src/amo/components/AddonBadges/index.js

```diff
@@ -35,27 +35,31 @@ type InternalProps = {|
 
 export class AddonBadgesBase extends React.Component<InternalProps> {
   static defaultProps: DefaultProps = {
-    _getPromotedCategory: getPromotedCategory,
+    _getPromotedCategories: getPromotedCategories,
   };
 
   render(): null | React.Node {
-    const { _getPromotedCategory, addon, clientApp, i18n } = this.props;
+    const { _getPromotedCategories, addon, clientApp, i18n } = this.props;
 
     if (!addon) {
       return null;
     }
 
-    const promotedCategory = _getPromotedCategory({
+    const promotedCategories = _getPromotedCategories({
       addon,
       clientApp,
       forBadging: true,
     });
 
     return (
       <div className="AddonBadges">
```

Comment:
Can you add a screenshot with the maximum realistic number of categories.. based on addons-server can we have 1,2,3,20 categories? How would that look in the UI? I'm surprised to see 0 css changes given this type of patch.

Reply:
There's 4 that show up as a badge -- Line, recommended, spotlight, and strategic.

![image](https://github.com/user-attachments/assets/3c44a8f2-9cee-44ea-8069-c1e04a0bbd73)

Though, now that I'm looking at this, I'm actually not sure if those blacked out recommended badges (spotlight and strategic) are meant to be revealed.

I don't recall (off the top of my head, at least) any existing checks to avoid Spotlight or Strategic badges (both from addons-server passing them as categories, nor from frontend), and there's also this mozilla/addons#14987 issue that seems to relate to this... @diox could you shed some light on this?

>I'm surprised to see 0 css changes given this type of patch.

The AddonBadges div also already holds multiple badges (not exclusively for promoted, ex. the "Available on Firefox for Android" badge), which is why its able to handle this change without any css edits. Whether it actually looks good in the case where every single badge is available, well...

![image](https://github.com/user-attachments/assets/6a4393b9-9d38-4bef-90cc-cba8cfbae0f8)

Reply:
Right, the second case is what I was after. From a UX perspective, what is the maximum amount of content, and does it look good. Let's ask Nicole and yes, we should either add svgs for the blacked out badges, or filter them.

Also, why do they all say recommended? I would expect we should name them appropriately as well..